### PR TITLE
Kubeconfig field added to provisioning request

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

As a part of PoC we extend provisioning request with field (optional) containing encoded kubeconfig.
It should facilitate provisioning Kyma on cluster provided by the caller.

Changes proposed in this pull request:

- DTO with Kubeconfig field

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/broker/instance_create_test.go [hunk 1]
```diff
@@ -116,6 +117,82 @@ func TestProvision_Provision(t *testing.T) {
...
+		assert.Equal(t, operation.InstanceID, instanceID)
+
+		assert.Equal(t, globalAccountID, operation.ProvisioningParameters.ErsContext.GlobalAccountID)
+		assert.Equal(t, clusterName, operation.ProvisioningParameters.Parameters.Name)
+		assert.Equal(t, userID, operation.ProvisioningParameters.ErsContext.UserID)
+		assert.Equal(t, "req-region", operation.ProvisioningParameters.PlatformRegion)
+
+		kubeconfigParameter, err := base64.StdEncoding.DecodeString(operation.ProvisioningParameters.Parameters.Kubeconfig)
+		require.NoError(t, err)
+		assert.Regexp(t, `.*kind: Config$`, string(kubeconfigParameter))
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/broker/instance_create_test.go:183]: nit: maybe we can match the string exactly to line 155 where it's defined?
https://github.com/kyma-project/control-plane/blob/518843eff040e1e0300c9ae1c5da5feae1f2793a/components/kyma-environment-broker/internal/broker/instance_create_test.go#L155

I think KEB should treat the user provided `kubeconfig` as pretty much opaque slice of bytes so the content in `operation` should be equal to provisioning payload.

**Review by wozniakjan — COMMENTED** [review]
lgtm, please consider few nits below. Also consider `update` endpoint, kubeconfigs can have certs/tokens and those can expire, users may need a way to update the kubeconfigs later.

File: components/kyma-environment-broker/internal/broker/instance_create_test.go [hunk 2]
```diff
@@ -116,6 +117,82 @@ func TestProvision_Provision(t *testing.T) {
...
+			ServiceID:     serviceID,
+			PlanID:        planID,
+			RawParameters: json.RawMessage(fmt.Sprintf(`{"name": "%s", "kubeconfig": "%s"}`, clusterName, kubeconfigEncoded)),
+			RawContext:    json.RawMessage(fmt.Sprintf(`{"globalaccount_id": "%s", "subaccount_id": "%s", "user_id": "%s"}`, globalAccountID, subAccountID, "Test@Test.pl")),
+		}, true)
+		t.Logf("%+v\n", *provisionEndpoint)
+
+		// then
+		require.NoError(t, err)
+		assert.Regexp(t, "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$", response.OperationData)
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create_test.go:166]: This was just copied from other tests case. But you are absolutely right.

File: components/kyma-environment-broker/internal/broker/instance_create_test.go (hunk 1 above)
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create_test.go:183]: :+1:

**Comment by jaroslaw-pieszka** [comment]
> lgtm, please consider few nits below. Also consider `update` endpoint, kubeconfigs can have certs/tokens and those can expire, users may need a way to update the kubeconfigs later.

This was supposed to be PoC - so with restricted validation (or no validation), not considering update yet.
But good point.

File: components/kyma-environment-broker/internal/broker/instance_create_test.go (hunk 1 above)
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create_test.go:183]: In the next commit.

File: components/kyma-environment-broker/internal/broker/instance_create_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create_test.go:166]: I need to figure out how it deviates from UUID.

File: resources/kcp/charts/kyma-environment-broker/values.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ global:
   images:
     kyma_environment_broker:
       dir:
-      version: "PR-1991"
+      version: "PR-2000"
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply resources/kcp/charts/kyma-environment-broker/values.yaml:5]: I recall the millennial bug and code 2000 resistant. Do we have a problem here? ;)

File: components/kyma-environment-broker/internal/broker/instance_create_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_create_test.go:168]: Comment added.

**Review by wozniakjan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**